src/observer/index.js

```diff
@@ -172,11 +173,27 @@ function copyAugment (target, src, keys) {
 
 function defineReactive (obj, key, val) {
   var dep = new Dep()
-  var childOb = Observer.create(val)
+
+  var target = {
+    val: val
+  }
+
+  if (config.convertAllProperties) {
+    var property = Object.getOwnPropertyDescriptor(obj, key)
+    if (property && property.get && property.set) {
```

Comment:
No worries it's all good feedback. 

I think the getter/setter stuff makes sense, but might as well also make it safely handle 'configurable' being false. For now I'll make it so it just leaves everything alone; let me know if you think some other behavior makes more sense.